spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -436,7 +436,7 @@ For example, a repository interface may inherit the `CrudRepository` interface d
 
 Repositories perform operations on entities. For repository methods that are annotated with `@Insert`, `@Update`, `@Save`, or `@Delete`, the entity type is determined from the method parameter type.  For `find` and `delete` methods where the return type is an entity, array of entity, or parameterized type such as `List<MyEntity>` or `Page<MyEntity>`, the entity type is determined from the method return type.  For `count`, `exists`, and other `delete` methods that do not return the entity or accept the entity as a parameter, the entity type cannot be determined from the method signature and a primary entity type must be defined for the repository.
 
-Users of Jakarta Data declare a primary entity type for a repository by inheriting from a built-in repository super interface, such as `BasicRepository`, and specifying the primary entity type as the first type variable.  The primary entity type is assumed for methods that do not otherwise specify an entity type, such as `countByPriceLessThan`. Methods that require a primary entity type raise `MappingException` if a primary entity type is not provided.
+Users of Jakarta Data declare a primary entity type for a repository by inheriting from a built-in repository super interface, such as `BasicRepository`, and specifying the primary entity type as the first type variable. For repositories that do not inherit from a super interface with a type parameter to indicate the primary entity type, life cycle methods on the repository determine the primary entity type if all life cycle methods where the method parameter is a type, an array of type, or is parameterized with a type that is annotated as an entity, correspond to the same entity type. The primary entity type is assumed for methods that do not otherwise specify an entity type, such as `countByPriceLessThan`. Methods that require a primary entity type raise `MappingException` if a primary entity type is not provided.
```

Comment:
```suggestion
Users of Jakarta Data declare a primary entity type for a repository by inheriting from a built-in repository super interface, such as `BasicRepository`, and specifying the primary entity type as the first type variable. For repositories that do not inherit from a super interface with a type parameter to indicate the primary entity type, life cycle methods on the repository determine the primary entity type. To do so, all life cycle methods where the method parameter is a type, an array of type, or is parameterized with a type that is annotated as an entity, must correspond to the same entity type. The primary entity type is assumed for methods that do not otherwise specify an entity type, such as `countByPriceLessThan`. Methods that require a primary entity type raise `MappingException` if a primary entity type is not provided.
```
Took me a few reads of this sentence to understand it, so splitting it in to might help. Also, what happens if a User doesn't inherit from a built-in repository interface, and also doesn't use the same entity type? Presumably an exception, but should we specify what kind?

Reply:
That's a good suggestion to improve the clarify. I'll add it.
Regarding your question,
> Also, what happens if a User doesn't inherit from a built-in repository interface, and also doesn't use the same entity type? Presumably an exception, but should we specify what kind?

That is covered further down in documentation that was already there before this pull. It states:
"Methods that require a primary entity type raise `MappingException` if a primary entity type is not provided."